src/comps/CartTab.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { List, Button, Typography, Box, Snackbar, Alert, LinearProgress } from '@mui/material';
+
+import { removeFromCart, selectCart, selectCartTotalPrice } from '../reducers/cartReducer';
+import { buyProduct, selectUserCredit } from '../reducers/userReducer'
+import CartItem from './CartItem';
+import Product from '../types/Product';
+import OrderCompleteDialog from './OrderCompleteDialog';
+
+const CartTab: React.FC = () => {
+  const dispatch = useDispatch();
+  const cart = useSelector(selectCart);
+  const totalPrice = useSelector(selectCartTotalPrice);
+  const userCredit = useSelector(selectUserCredit);
+
+  const isCartEmpty: boolean = cart.length === 0;
+
+  const [orderSuccessOpen, setOrderSuccessOpen] = useState(false);
```

Comment:
```suggestion
  const [orderSuccessOpen, setOrderSuccessOpen] = useState<boolean>(false);
```
also for the rest of the code